driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -370,8 +370,8 @@ public void testGetResumeTokenReturnsPostBatchResumeTokenAfterGetMore()
             // use reflection to access the postBatchResumeToken
             AggregateResponseBatchCursor<?> batchCursor = getBatchCursor(cursor);
 
-            // check equality in the case where the batch has not been iterated at all
-            assertEquals(cursor.getResumeToken(), batchCursor.getPostBatchResumeToken());
+            // resume token should be null before iteration
+            assertNull(cursor.getResumeToken());
```

Comment:
This test was wrong.  Before the bug fix, both the values were null.  Afterwards, the postBatchResumeToken is set on the batch cursor but not the high-level cursor.

Reply:
Should you add a `assertNotNull(batchCursor.getPostBatchResumeToken())` check as well?

Reply:
Done